src/SlackConnector/SlackConnection.cs

```diff
@@ -63,20 +63,26 @@ public void Initialise(ConnectionInformation connectionInformation)
             ConnectedSince = DateTime.Now;
         }
 
-        private async Task ListenTo(InboundMessage inboundMessage)
+        private Task ListenTo(InboundMessage inboundMessage)
         {
-            if (inboundMessage?.MessageType != MessageType.Message)
-                return;
-            if (string.IsNullOrEmpty(inboundMessage.User))
-                return;
-            if (!string.IsNullOrEmpty(Self.Id) && inboundMessage.User == Self.Id)
```

Comment:
they are moved into HandleMessage, as not all messages have a user associated with

https://github.com/enyim/SlackConnector/blob/976e807d564b7a0374da49b472f831d3089b3ac8/src/SlackConnector/SlackConnection.cs#L82